packages/xstate-test/src/utils.ts

```diff
@@ -78,6 +78,12 @@ export function formatPathTestResult(
 export function getDescription<T, TContext>(state: AnyState): string {
   const contextString =
     state.context === undefined ? '' : `(${JSON.stringify(state.context)})`;
+  const transitionDescription = state.transitions[0].description
+    ? state.transitions[0].description.trim()
+    : '';
```

Comment:
I wonder if this is actually correct.

`getDescription` is actually kinda like `getStateDescription` and you are mixing descriptions of a transition with it. The problem I'm seeing here is that you will only ever print a description of **the very last** transition and that you will only print a description for a single transition but in reality, multiple transitions can be available on that `state` that you have access to here.

The first problem is especially important here because this description won't ever be printed anywhere and it seems off to me that only the last transition is ever taken into consideration:
```js
      const machine = createTestMachine({
        initial: "a",
        states: {
          a: {
            on: {
              EVENT: { description: description, target: "b" },
            },
          },
          b: {
            on: {
              EVENT: { target: "c" },
            },
          },
          c: {},
        },
      });
```

Reply:
I'm not sure I follow what you're saying. I think what you are saying is

`getDescription` generating a description from the final state only. So in order to do this, what we'll need is to evaluate if this event has been called along the path to get here. As such, what we really want is something more like

```
// generate the test messge
// check if that path used our event
// if it did, append any events with a description onto the test message
```

That's a lot more than I considered 🤣 . I can try to look through this code and see if there's a way I can do that.